src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -2003,6 +2005,20 @@ public static AccessMode valueFromMethodName(String methodName) {
                 default -> throw new IllegalArgumentException("No AccessMode value for method name " + methodName);
             };
         }
+
+        // Mapping from the ordinal to AccessMode
+        private static final Map<Integer, AccessMode> modeToAccessMode = initOrdinalToAccessModeMap();
+        private static Map<Integer, AccessMode> initOrdinalToAccessModeMap() {
+            Map<Integer, AccessMode> map = new HashMap<>();
+            for (AccessMode am : AccessMode.values()) {
+                map.put(am.ordinal(), am);
+            }
+            return map;
+        }
+
+        static AccessMode valueFromOrdinal(int mode) {
+            return modeToAccessMode.get(mode);
```

Comment:
Can't this be simplified to `AccessMode.values()[mode]` since this is only rarely used in the exception logic?

If we do need a cache, I would recommend a stable array like
```java
private static final @Stable AccessMode[] VALUES = values();
```
and users call this accessor instead.

The code in `getMethodHandleUncached` can benefit from this caching mechanism.

Reply:
Good suggestion.